src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts

```diff
@@ -120,48 +129,119 @@ export class StudentExamDetailComponent implements OnInit {
+    }
+
+    /**
+     * Updates the working time duration values of the form whenever the percent value has been changed by the user.
+     */
+    updateWorkingTimeDuration() {
+        const seconds = Math.round(this.normalWorkingTime * (1.0 + this.workingTimeFormValues.percent / 100));
+        this.setWorkingTimeDuration(seconds);
+    }
+
+    /**
+     * Updates the hours, minutes, and seconds values of the form.
+     * @param seconds the total number of seconds of working time.
+     * @private
+     */
+    private setWorkingTimeDuration(seconds: number) {
+        const workingTime = this.artemisDurationFromSecondsPipe.secondsToDuration(seconds);
+        this.workingTimeFormValues.hours = workingTime.days * 24 + workingTime.hours;
+        this.workingTimeFormValues.minutes = workingTime.minutes;
+        this.workingTimeFormValues.seconds = workingTime.seconds;
+    }
+
+    /**
+     * Uses the current durations saved in the form to update the extension percent value.
+     */
+    updateWorkingTimePercent() {
+        this.workingTimeFormValues.percent = this.getWorkingTimePercentDifference();
+    }
+
+    /**
+     * Calculates how many seconds the currently set working time has in total.
+     * @private
+     */
+    private getWorkingTimeSeconds(): number {
+        return this.workingTimeFormValues.hours * 3600 + this.workingTimeFormValues.minutes * 60 + this.workingTimeFormValues.seconds;
     }
 
-    examIsVisible(): boolean {
+    /**
+     * Calculates the difference in whole percent between the regular working time and the currently set individual working time.
+     *
+     * E.g., with a regular time of "1h" and a current value of "1h30min" returns 50.
+     * @private
+     */
+    private getWorkingTimePercentDifference(): number {
+        return Math.round((this.getWorkingTimeSeconds() / this.normalWorkingTime - 1.0) * 100);
+    }
+
+    /**
+     * Checks if the user should be able to edit the inputs.
+     */
+    isFormDisabled(): boolean {
+        return this.isSavingWorkingTime || !this.canChangeExamWorkingTime();
+    }
+
+    /**
+     * Checks if the working time of the exam can still be changed.
+     * @private
+     */
+    private canChangeExamWorkingTime(): boolean {
         if (this.isTestRun) {
-            // for test runs we always want to be able to change the working time
-            return !!this.studentExam.submitted;
+            // for unsubmitted test runs we always want to be able to change the working time
+            return !this.studentExam.submitted;
         } else if (this.studentExam.exam) {
-            // Disable the form to edit the working time if the exam is already visible
-            return dayjs(this.studentExam.exam.visibleDate).isBefore(dayjs());
+            // for student exams it can only be changed before the student is able to see it
+            return dayjs().isBefore(dayjs(this.studentExam.exam.visibleDate));
+        } else {
```

Comment:
I think we could omit this else block and just return

Reply:
I intentionally used the `else`-Block as this avoids the early returns in the middle of the method and instead keeps the return as the last statement in every case.

Reply:
I see the `else` as an explicit fallback instead of an implicit one caused by returning early from the function from the `else if`. Using it therefore makes the control flow more explicit, which is why I prefer it.

Reply:
I understand. As you agreed to revert it in your other PR, I think it makes sense to do it the implicit way, as you call it, as well.
To give you some numbers, doing some regex search within the project, it appears that the implicit style gets used ten times more than the explicit style (of course I didn't check every result but the direction should be correct).

I personally just like to have the codebase using the same style to make it easier to read the code (and of course, I like my style better, I won't deny it :eyes:).

Reply:
I understand. As you agreed to revert it in your other PR, I think it makes sense to do it the implicit way, as you call it, as well.
To give you some numbers, doing some regex search within the project, it appears that the implicit style gets used ten times more than the explicit style (of course I didn't check every result but the direction should be correct).

I personally just like to have the codebase using the same style to make it easier to read the code (and of course, I like my style better, I won't deny it 👀).